### PR TITLE
Ansible UI fixes

### PR DESCRIPTION
**Pull request by adarshdubey-star**
`Test results`

pytest tests/foreman/ui/test_ansible.py 
============================================================================ test session starts ============================================================================
platform linux -- Python 3.8.6, pytest-7.1.3, pluggy-1.0.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/addubey/work/robottelo, configfile: pyproject.toml
plugins: forked-1.4.0, services-2.2.1, cov-3.0.0, xdist-3.1.0, ibutsu-2.2.4, reportportal-5.1.3, mock-3.10.0
collected 3 items                                                                                                                                                           

tests/foreman/ui/test_ansible.py ...                                                                                                                                  [100%]

### CODE REVIEW

File: airgun/views/ansible_role.py [hunk 1]
```diff
@@ -21,10 +21,10 @@ class AnsibleRolesView(BaseLoggedInView, SearchableViewMixin):
     is present, without the search widget or table.
     """
 
-    title = Text("//h1")
+    title = Text("//h1[text()='Ansible Roles']")
```
Comment (shweta83) [inline airgun/views/ansible_role.py:24]: Can we avoid the exact string match as if anything changes in string in future, then it doesn't cause failure.

File: airgun/entities/ansible_role.py [hunk 2]
```diff
@@ -39,10 +39,9 @@ def imported_roles_count(self):
         view = self.navigate_to(self, 'All')
         # Before any roles have been imported, no table or pagination widget are
         # present on the page
-        if not view.pagination.is_displayed:
-            return 0
-        else:
-            return int(view.total_imported_roles.read())
+        # Applying wait_displayed for the page to get rendered
+        view.wait_displayed()
```
Comment (shweta83) [inline airgun/entities/ansible_role.py:43]: @adarshdubey-star Any specific reason to use `wait_displayed` as we can use `is_displayed` here as well.

**Review by shweta83 — CHANGES_REQUESTED** [review]

File: airgun/entities/ansible_role.py (hunk 2 above)
Comment (adarshdubey-star) [reply airgun/entities/ansible_role.py:43]: Yeah the issue was the page was not displayed and it returned 0 with the previous logic making the assertion fail, just gave some more time to get it rendered properly.

File: airgun/views/ansible_role.py (hunk 1 above)
Comment (adarshdubey-star) [reply airgun/views/ansible_role.py:24]: Makes sense!

File: airgun/views/ansible_variable.py [hunk 3]
```diff
@@ -17,9 +17,9 @@
 class AnsibleVariablesView(BaseLoggedInView, SearchableViewMixin):
     """Main Ansible Variables view"""
 
-    title = Text("//h1[contains(., text()='Ansible Variables')")
+    title = Text("//h1[text()='Ansible Variables']")
```
[the remark this answers is not in the document]
Comment (adarshdubey-star) [reply airgun/views/ansible_variable.py:20]: yeah!

File: airgun/views/ansible_role.py [hunk 4]
```diff
@@ -21,10 +21,10 @@ class AnsibleRolesView(BaseLoggedInView, SearchableViewMixin):
     is present, without the search widget or table.
     """
 
-    title = Text("//h1")
+    title = Text("//h1[text()]")
```
Comment (omkarkhatavkar) [inline airgun/views/ansible_role.py:24]: why are using the `text()` here ?

Comment (adarshdubey-star) [reply airgun/views/ansible_role.py:24]: Just to be a little precise I was matching the exact string but I think the tag itself will do the job please correct me here @omkarkhatavkar

Comment (omkarkhatavkar) [reply airgun/views/ansible_role.py:24]: I don't this it is required there

Comment (adarshdubey-star) [reply airgun/views/ansible_role.py]: updated

**Review by omkarkhatavkar — APPROVED** [review]

File: airgun/entities/ansible_role.py (hunk 2 above)
Comment (shweta83) [reply airgun/entities/ansible_role.py:43]: This works as well.

File: airgun/views/ansible_variable.py [hunk 5]
```diff
@@ -17,9 +17,9 @@
 class AnsibleVariablesView(BaseLoggedInView, SearchableViewMixin):
     """Main Ansible Variables view"""
 
-    title = Text("//h1[contains(., text()='Ansible Variables')")
+    title = Text("//h1")
```
Comment (shweta83) [inline airgun/views/ansible_variable.py:20]: This works now but can we add  "Text("//h1[contains(normalize-space(.)='Ansible Variables')")". What do you say @omkarkhatavkar ?

Comment (omkarkhatavkar) [reply airgun/views/ansible_variable.py:20]: Yes, we can us the normalised space, the earlier locator itself was wrong. @shweta83 you shared also needs to correct. It should be `"//h1[contains(normalize-space(.),'Ansible Variable')]"`

Comment (shweta83) [reply airgun/views/ansible_variable.py:20]: Thanks @omkarkhatavkar.

**Review by synkd — APPROVED** [review]

File: airgun/views/ansible_variable.py (hunk 5 above)
Comment (adarshdubey-star) [reply airgun/views/ansible_variable.py]: Updated @omkarkhatavkar @shweta83

**Review by shweta83 — APPROVED** [review]
Ack

**Pull request merged**